src/main/scala/zio/prelude/AddMultiplyShape.scala

```diff
@@ -0,0 +1,100 @@
+
+  def Addition: Addition[Sum[A]]
+
+  def Multiplication: Multiplication[Prod[A]]
+}
+
+object AddMultiplyShape {
+
+  /// Helper classes to make the code shorter, but we don't want them to be exposed to ZIO Prelude users
+  private abstract class Ring[A]
+      extends AnnihilatingZero[A, Ring.Addition, Ring.Multiplication]
+      with DistributiveMultiply[A, Ring.Addition, Ring.Multiplication]
+      with SubtractShape[A, Ring.Addition, Ring.Multiplication]
+
+  private object Ring {
+    type Addition[x]       = Commutative[x] with Inverse[x]
+    type Multiplication[x] = Commutative[x] with Identity[x]
+  }
+
+  private abstract class Field[A]
+      extends AnnihilatingZero[A, Field.Addition, Field.Multiplication]
+      with DistributiveMultiply[A, Field.Addition, Field.Multiplication]
+      with DivideShape[A, Field.Addition, Field.Multiplication]
+      with SubtractShape[A, Field.Addition, Field.Multiplication]
+
+  private object Field {
+    type Addition[x]       = Commutative[x] with Inverse[x]
+    type Multiplication[x] = Commutative[x] with InverseNonZero[x]
+  }
+
+  implicit val IntAnnihilatingZeroDistributiveMultiply: classic.Ring[Int] = new Ring[Int] {
+    override def add(l: => Int, r: => Int): Int      = l + r
+    override def multiply(l: => Int, r: => Int): Int = l * r
+    override def subtract(l: => Int, r: => Int): Int = l - r
+
+    override def Addition: Commutative[Sum[Int]] with Inverse[Sum[Int]] =
+      Associative.IntSumCommutativeInverse
+
+    override def Multiplication: Commutative[Prod[Int]] with Identity[Prod[Int]] =
+      Associative.IntProdCommutativeIdentity
+  }
+
+  implicit val DoubleAnnihilatingZeroDistributiveMultiply: classic.Field[Double] = new Field[Double] {
+    override def add(l: => Double, r: => Double): Double      = l + r
+    override def divide(l: => Double, r: => Double): Double   = l / r
+    override def multiply(l: => Double, r: => Double): Double = l * r
+    override def subtract(l: => Double, r: => Double): Double = l - r
+
+    override def Addition: Commutative[Sum[Double]] with Inverse[Sum[Double]] =
+      Associative.DoubleSumCommutativeInverse
+
+    override def Multiplication: Commutative[Prod[Double]] with InverseNonZero[Prod[Double]] =
+      Associative.DoubleProdCommutativeIdentity
+  }
+}
+
+trait AddMultiplyShapeSyntax {
+
+  /**
+   * Provides infix syntax for adding or multiplying two values.
+   */
+  implicit class AddMultiplyShapeOps[A](private val l: A) {
+
+    /**
+     * A symbolic alias for `add`.
+     */
+    def +++(r: => A)(implicit associative: AddMultiplyShape[A, Associative, Associative]): A =
+      associative.add(l, r)
+
+    /**
+     * Add two values.
+     */
+    def add(r: => A)(implicit associative: AddMultiplyShape[A, Associative, Associative]): A =
+      associative.add(l, r)
+
+    /**
+     * A symbolic alias for `multiply`.
+     */
+    def ***(r: => A)(implicit associative: AddMultiplyShape[A, Associative, Associative]): A =
```

Comment:
There's not. This was just my invention, because mere `+` didn't work. So that's why I opted for a triplet `+++`. `*` works fine, but I wanted the symbols to be "symmetrical".
I suspect that I might have done some mistake. Do you know what I did wrong and/or how to make `+` work as an extension method?

Reply:
Btw the errors I get look plainly like this
```scala
[error] /home/ondra/Projects/zio-prelude/src/main/scala/zio/prelude/DistributiveMultiply.scala:23:21: value + is not a member of type parameter A
[error]         (a1 *** (a2 + a3)) <-> ((a1 *** a2) + (a1 *** a3))
[error]                     ^
[error] /home/ondra/Projects/zio-prelude/src/main/scala/zio/prelude/DistributiveMultiply.scala:23:45: value + is not a member of type parameter A
[error]         (a1 *** (a2 + a3)) <-> ((a1 *** a2) + (a1 *** a3))
[error]                                             ^
[error] /home/ondra/Projects/zio-prelude/src/main/scala/zio/prelude/DistributiveMultiply.scala:37:14: value + is not a member of type parameter A
[error]         ((a1 + a2) *** a3) <-> ((a1 *** a3) + (a2 *** a3))
[error]              ^
[error] /home/ondra/Projects/zio-prelude/src/main/scala/zio/prelude/DistributiveMultiply.scala:37:45: value + is not a member of type parameter A
[error]         ((a1 + a2) *** a3) <-> ((a1 *** a3) + (a2 *** a3))
[error]                                             ^
[error] four errors found
```